### PR TITLE
Proxy generation Part 1

### PR DESCRIPTION
**Pull request by moozzyk**
Proxy generation Part 1. Contains
- dotnet signalr tool
- metadata based hub discovery

### CODE REVIEW

File: build/dependencies.props [hunk 1]
```diff
@@ -13,6 +13,7 @@
     <TestSdkVersion>15.3.0-*</TestSdkVersion>
     <XunitVersion>2.3.0-beta2-*</XunitVersion>
     <RxVersion>3.1.1</RxVersion>
+    <MonoCecilVersion>0.10.0-beta6</MonoCecilVersion>
```
Comment (moozzyk) [inline build/dependencies.props]: @Eilon  - Can we get this approved?

File: src/Microsoft.AspNetCore.SignalR.Tools/HubDiscovery.cs [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+                                var assemblyDefintion = ModuleDefinition.ReadModule(fullPath, parameters).Assembly;
+                                _assemblyCache[assemblyReferenceName.FullName] = assemblyDefintion;
+                                return assemblyDefintion;
+                            }
+                            catch
+                            {
+                                // Unable to read module
+                                // TODO: Log?
+                            }
+                        }
```
Comment (mikaelm12) [inline src/Microsoft.AspNetCore.SignalR.Tools/HubDiscovery.cs]: Remove new line

**Comment by muratg** [comment]
@moozzyk what are we doing about this PR? It's quite stale.

**Comment by moozzyk** [comment]
We need to decide if we want to move forward with this feature in general and if we want (and can) use Roslyn instead of Cecil.

**Comment by muratg** [comment]
I think the feature would be valuable.  @davidfowl thoughts?

**Comment by davidfowl** [comment]
Yes, we still want to do this

**Comment by Eilon** [comment]
I would strongly prefer if we use Roslyn and not Mono Cecil because it's easier for us to bundle everything up to ship.

**Comment by davidfowl** [comment]
We may as well use reflection at this point. Run like an ef tool.

**Pull request closed without merging**